Jenkinsfile

```
@@ -58,21 +58,53 @@ node('linux') {
 
     if (infra.isTrusted()){
         stage('Publish on Azure') {
-            /* -> https://github.com/Azure/blobxfer
-            Require credential 'JAVADOC_STORAGEACCOUNTKEY' set to the storage account key */
-            withCredentials([string(credentialsId: 'JAVADOC_STORAGEACCOUNTKEY', variable: 'JAVADOC_STORAGEACCOUNTKEY')]) {
-                sh './scripts/blobxfer upload \
-                --local-path /data/site \
-                --storage-account-key $JAVADOC_STORAGEACCOUNTKEY \
-                --storage-account prodjavadoc \
-                --remote-path javadoc \
-                --recursive \
-                --mode file \
-                --skip-on-md5-match \
-                --file-md5 \
-                --connect-timeout 30 \
-                --delete'
-            }
+            parallel(
+                failFast: false,
+                'With blobxfer on prodjavadoc': {
+                    /* -> https://github.com/Azure/blobxfer
+                    Require credential 'JAVADOC_STORAGEACCOUNTKEY' set to the storage account key */
+                    withCredentials([string(credentialsId: 'JAVADOC_STORAGEACCOUNTKEY', variable: 'JAVADOC_STORAGEACCOUNTKEY')]) {
+                        sh './scripts/blobxfer upload \
+                        --local-path /data/site \
+                        --storage-account-key $JAVADOC_STORAGEACCOUNTKEY \
+                        --storage-account prodjavadoc \
+                        --remote-path javadoc \
+                        --recursive \
+                        --mode file \
+                        --skip-on-md5-match \
+                        --file-md5 \
+                        --connect-timeout 30 \
+                        --delete'
+                    }
+                },
+                'With azcopy on javadocjenkinsio': {
+                    infra.withFileShareServicePrincipal([
+                        servicePrincipalCredentialsId: 'trustedci_javadocjenkinsio_fileshare_serviceprincipal_writer',
+                        fileShare: 'javadoc-jenkins-io',
+                        fileShareStorageAccount: 'javadocjenkinsio',
+                        durationInMinute: 45
```

```suggestion
                        durationInMinute: 20
```
Upload time went from 42 minutes with blobxfer to 14 minutes with azcopy 🥳